test/parallel/test-dns-channel-cancel.js

```diff
@@ -14,7 +14,7 @@ const addMessageListener = () => {
   server.removeAllListeners('message');
 
   server.once('message', () => {
-    server.once('message', common.mustNotCall);
+    server.once('message', common.mustNotCall());
```

Comment:
The DNS tests fail now, which I assume is due to an existing bug (presumably in the test). Are DNS requests sent concurrently so that multiple messages might be in-flight when `resolver.cancel()` is called? Or are queued dgram writes not cancelled? cc @addaleax @szmarczak.

Reply:
I think you're right. I'm not sure if something like this would fix that:
```suggestion
    queueMicrotask(() => {
      server.once('message', common.mustNotCall());
    });
```

Reply:
but then the microtask needs to be cancelled when another `addMessageListener` is called. I'll test this out.

Reply:
Tracked in https://github.com/nodejs/node/issues/44428 due to inactivity here.

Reply:
I apologize for the delay @tniessen! I opened #44432 PR to fix the DNS issue. I can't remember why I put `mustNotCall` there, I guess I misunderstood something back then.